File: OloEditor/SandboxProject/Assets/Scripts/Source/PerfBob.cs
using System;

using OloEngine;

namespace Sandbox
{
	// Minimal per-frame scripting workload for the scripts_swarm_cs perf stress
	// scene (generate_perf_scenes.py): one Translation get + set per tick, so
	// N entities measure the Mono interop crossing cost, not script body cost.
	public class PerfBob : Entity
	{
		private float m_Time;

		void OnUpdate(float ts)
		{
			m_Time += ts;
			Vector3 translation = Translation;
			translation.Y += (float)Math.Sin(m_Time * 2.0f) * ts * 0.5f;
			Translation = translation;
		}
	}
}

File: OloEditor/SandboxProject/Assets/Scripts/LuaScripts/PerfBob.lua
-- PerfBob.lua — minimal per-frame scripting workload for the scripts_swarm
-- perf stress scenes (generate_perf_scenes.py). Each tick does one
-- get_translation + set_translation round trip, so N entities measure the
-- Lua<->engine interop crossing cost, not script body cost.

local PerfBob = {}

local t = 0.0

function PerfBob.OnUpdate(id, dt)
    t = t + dt
    local pos = entity_utils.get_translation(id)
    pos.y = pos.y + math.sin(t * 2.0) * dt * 0.5
    entity_utils.set_translation(id, pos)
end

return PerfBob
